sdk/python/feast/feature_source.py

```diff
@@ -0,0 +1,532 @@
+# Copyright 2020 The Feast Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import enum
+from typing import MutableMapping
+
+from feast.core.FeatureSource_pb2 import FeatureSourceSpec as FeatureSourceSpecProto
+
+
+class SourceType(enum.Enum):
+    """
+    FeatureSource value type. Used to define source types in FeatureSource.
+    """
+
+    UNKNOWN = 0
+    BATCH_FILE = 1
+    BATCH_BIGQUERY = 2
+    STREAM_KAFKA = 3
+    STREAM_KINESIS = 4
+
+
+class FileFormat(enum.Enum):
+    """
+    FileOptions value type. Used to define source types in FeatureSource.
+    """
+
+    UNKNOWN = 0
+    PARQUET = 1
+    AVRO = 2
+
+
+class FeatureSourceSpecOptions:
```

Comment:
What is the point of this class?

Reply:
Not much other than to state the need to create `to_proto` and `from_proto` method.

Reply:
Doesnt seem useful.

Reply:
Removed.